current_fall2025/scripts/app.py

```
@@ -30,170 +80,190 @@ def load_llm():
         try:
             metadata = doc.metadata
             source_id = metadata.get("source", "Unknown")
             title = metadata.get("title_info_primary_tsi", "Untitled")
             doc_url = f"https://www.digitalcommonwealth.org/search/{source_id}"
+            
             with st.expander(f"📄 {title} (ID: {source_id})", expanded=False):
-                st.markdown(f"**Preview:** {doc.page_content[:300]}...")
+                content_preview = doc.page_content[:300] + "..." if doc.page_content else "No text content available."
+                st.markdown(f"**Preview:** {content_preview}")
                 st.markdown(f"[🔗 View Original Source]({doc_url})")
         except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)